controllers/storagecluster/uninstall_reconciler.go

```diff
@@ -311,6 +311,12 @@ func (r *StorageClusterReconciler) deleteResources(sc *ocsv1.StorageCluster) err
 		return err
 	}
 
+	kmsRMgr := &kmsResourceManager{}
```

Comment:
This doesn't look right to me. Why can't we have the kmsResourceManager in the list of objects between lines 290 and 299?

If the deletion should happen only after ceph resources are deleted, you can add kmsResourceManager after ocsCephBlockPools.

Reply:
+1

Reply:
The main reason we cannot add the `kmsResourceManager` into the array is, they are a list of `resourceManager` inerface objects. `kmsResourceManager` has only one method `ensureDeleted()` as we are not creating anything it don't have `ensureCreated()` method.

We could add a dummy `ensureCreated()` function and make `kmsResourceManager` object compliant to the `resourceManager` interface

Reply:
If we are not creating it then who is creating it? Is rook creating it? if yes then we should delete it from the rook itself. In general, we should not delete such objects which we are not creating.

Either we can create a dummy `ensureCreated()` or we can change the `ensureDeleted` to `deleteKms*` which will be a method on the `StorageClusterReconciler` itself.

Reply:
It is the console UI (while creating StorageCluster) which takes the input from user and creates these KMS ConfigMaps and SecretToken resources. But while removing/uninstalling the StorageCluster, we no longer require these resources and we could safely remove them...

Reply:
In that case, I say go with Nitin's suggestion to just make an explicit `deleteKms*()` function rather than creating a custom struct.